### PR TITLE
Add JavaScript coverage reporting

### PR DESCRIPTION
**Pull request by cafca**
Adds JavaScript coverage reporting on codecov.io

## 📋 Checklist

- [ ] ~~Add tests that cover your changes~~
- [x] Add this PR to the _Unreleased_ section in `CHANGELOG.md`
- [ ] ~~Link this PR to any issues it closes~~
- [ ] ~~New files contain a SPDX license header~~
- [x] Check if descriptions and terminology match `handbook` content (and visa-versa)

### CODE REVIEW

File: p2panda-js/jest.config.ts [hunk 1]
```diff
@@ -10,6 +10,9 @@ export default {
     wasm: '<rootDir>/wasm',
   },
   modulePathIgnorePatterns: ['<rootDir>/wasm-web'],
+  // Skip reporting coverage for auto-generated wasm module that should be
+  // tested from p2panda-rs.
+  coveragePathIgnorePatterns: ['<rootDir>/wasm'],
```
Comment (cafca) [inline p2panda-js/jest.config.ts:15]: Interesting question actually: What is our testing strategy for `wasm`?

File: .github/workflows/js.yml [hunk 2]
```diff
@@ -44,9 +44,12 @@ jobs:
         working-directory: ${{ env.working_directory }}
 
       - name: Run tests
-        run: npm run test
+        run: npm run test --coverage
         working-directory: ${{ env.working_directory }}
 
+      - if: matrix.node == 16
```
Comment (cafca) [inline .github/workflows/js.yml:50]: Coverage is only collected from one of the JavaScript runs.

File: p2panda-js/jest.config.ts (hunk 1 above)
Comment (sandreae) [reply p2panda-js/jest.config.ts:15]: This was one of my christmas PRs. We do have tests for the wasm target code in p2panda-rs now: https://github.com/p2panda/p2panda/pull/184

Comment (sandreae) [reply p2panda-js/jest.config.ts:15]: We use https://rustwasm.github.io/wasm-bindgen/wasm-bindgen-test/index.html

Comment (sandreae) [reply p2panda-js/jest.config.ts:15]: I couldn't figure out how to add the wasm-bindgen-tests to our test coverage though :-(

Comment (cafca) [reply p2panda-js/jest.config.ts:15]: What a meanie...

**Review by adzialocha — APPROVED** [review]

**Pull request merged**